### DISCUSSION TITLE
Using REGEX in Entity Ruler

### DISCUSSION THREAD

**Discussion by lili111888** [explosion/spaCy#8854] [Help: Coding & Implementations] [answered]
I tried to find a phone number in this format: "Phone number: 01322 248065 " and I want to use REGEX:

"\d{5}\s+\d{6}", but It didn't work.

Here is my code:

```
ruler = nlp.add_pipe("entity_ruler")
patterns =  [{'label': "PHONENUMBER", "pattern": [{"TEXT": {"REGEX":"\d{5}\s\d{6}"}}]}]
ruler.add_patterns(patterns)
a = "Hello my name is John Smith. Here is my phone number 4167775555. another number is 12345 123456 "
    
result = []

doc =nlp(a)
for ent in doc.ents:
    output = {}
    output["label"] = ent.label_
    output["text"] = ent.text
    output["start"] = ent.start_char
    output["end"] = ent.end_char
    result.append(output)
print(json.dumps(result))
```

Could you please show me how to fix my code so that I will get the correct result?

**Comment by polm** [comment, marked as answer]
We actually had another discussion about matching phone numbers recently #6935 . The Matcher is intended for situations where tokenization is relatively predictable, which isn't necessarily the case with phone numbers. In particular, each item in a Matcher pattern has to be one token, so using a regex that contains whitespace is not going to work with the default English tokenizer settings. 

In this situation I would skip using the Matcher and just run your regex against the whole sentence. Here's an example:

```
import spacy
import re
import json

nlp = spacy.blank("en")

text = "Hello my name is John Smith. Here is my phone number 4167775555. another number is 12345 123456 "
doc = nlp(text)
    
PHONE_PAT = "\d{5}\s\d{6}"

ents = []
for match in re.finditer(PHONE_PAT, doc.text):
    ent = doc.char_span(match.start(), match.end(), label="PHONE")
    ents.append(ent)

doc.ents = ents

result = []
for ent in doc.ents:
    output = {}
    output["label"] = ent.label_
    output["text"] = ent.text
    output["start"] = ent.start_char
    output["end"] = ent.end_char
    result.append(output)
print(json.dumps(result))
```

**Reply by lili111888** [reply]
Thank you, it works
